tests/atexit/host/host.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/host.h>
+#include <openenclave/internal/calls.h>
+#include <openenclave/internal/error.h>
+#include <openenclave/internal/tests.h>
+#include <unistd.h>
+#include <cstdio>
+#include <cstdlib>
+#include <thread>
+
+#include "atexit_u.h"
+
+#define OCALL_WITH_INCREASE_SINGLE 0
+#define OCALL_WITH_INCREASE_MULTIPLE 1
+#define OCALL_WITH_AN_ECALL 2
+
+oe_enclave_t* enclave = NULL;
+int global_var = 0;
+
+void global_variable_increase_ocall(void)
+{
+    global_var++;
+}
+
+void with_an_ecall_ocall(void)
+{
+    global_var += 3;
+    uint32_t magic = 2;
+    oe_result_t result = get_magic_ecall(enclave, &magic);
+    // should fail since enclave was destoryed
+    OE_TEST(result == OE_REENTRANT_ECALL);
```

Comment:
So I actually don't think this is failing because the enclave was destroyed. This is still during the atexit call so the enclave is still around.

This is failing because OE doesn't allow reentrant ecalls. That is, on a thread where an ecall has already been make, you cannot make an ocall and then another ecall.

This is still a good thing to test but I think the comment should be changed

Reply:
I have updated this comment. Thanks.